app/scripts/metamask-controller.js

```diff
@@ -2494,14 +2558,22 @@ export default class MetamaskController extends EventEmitter {
       verifyPassword: this.verifyPassword.bind(this),
 
       // network management
-      setProviderType:
-        networkController.setProviderType.bind(networkController),
+      setProviderType: (type) => {
+        // when using this format, type happens to be the same as the networkClientId...
+        this.selectedNetworkController.setNetworkClientIdForMetamask(type);
+        return this.networkController.setProviderType(type);
+      },
+      setActiveNetwork: (networkConfigurationId) => {
+        // shouldnt be required now that we are using the latest sel net ctrl.
```

Comment:
if not required, can we delete it?

Reply:
aah the comment is actually the part we done need anymmore =P